Vite plugin docs (#20481)
cloudflare/cloudflare-docs: src/content/docs/workers/vite-plugin/api.mdx

---
pcx_content_type: reference
title: API
head: []
sidebar:
  order: 3
description: Vite plugin API
---

> ### `cloudflare`

Reviewer: This is a bit similar to who you prefaced `PluginConfig` with `interface` below, rather than leaving it free-standing.

The `cloudflare` plugin should be included in the Vite `plugins` array:

```ts {4, 7}
// vite.config.ts

import { defineConfig } from "vite";
import { cloudflare } from "@cloudflare/vite-plugin";

export default defineConfig({
	plugins: [cloudflare()],
});
```

It accepts an optional `PluginConfig` parameter.

### `interface PluginConfig`

- `configPath?: string`

  An optional path to your Worker config file.
  By default, a `wrangler.toml`, `wrangler.json`, or `wrangler.jsonc` file in the root of your application will be used as the Worker config.

- `viteEnvironment?: { name?: string }`

  Optional Vite environment options.
  By default, the environment name is the Worker name with `-` characters replaced with `_`.
  Setting the name here will override this.
  A typical use case is setting `viteEnvironment: { name: "ssr" }` to apply the Worker to the SSR environment.

- `persistState?: boolean | { path: string }`

  An optional override for state persistence.

>   By default, state is persisted to `.wrangler/state` in a `v3` subdirectory.

Reviewer: why not just:

```suggestion
  By default, state is persisted to `.wrangler/state/v3`.
```

Author: I think I had it like that in a previous iteration but it was highlighted that if the user provides a custom `persistState.path` then a `v3` subdirectory will still be created inside it. I don't really mind either way though.

Reviewer: Oh yeah... I forgot. So then perhaps we should say that state is persisted to a `v3` subdirectory of the `persistState` directory. By default this is `.wrangler/state`?

Or perhaps just don't mention v3 at all? The persistence directory is supposed to be a black box to the user, they shouldn't be trying to read it.

  A custom `path` can be provided or, alternatively, persistence can be disabled by setting the value to `false`.

- `inspectorPort?: number | false`

>   An optional inspector port to use for debugging your Workers.
>   The default value is `9229`.
>   Setting this to `false` will disable the debugging inspector.

Reviewer: ```suggestion
  An optional override for debugging your Workers.
  The default is to enable the debugging inspector to listen on port `9229`.
  A custom port can be provided, or setting this to `false` will disable the debugging inspector.
```

- `auxiliaryWorkers?: Array<AuxiliaryWorkerConfig>`

>   An optional array of auxiliary Workers.

Reviewer: We should define or explain what we mean by auxiliary workers; I don't think this is a standard term for Cloudflare Workers.

  You can use [service bindings](/workers/runtime-apis/bindings/service-bindings/) to call auxiliary Workers from your main (entry) Worker.
  All requests are routed through your entry Worker.
  During the build, each Worker is output to a separate subdirectory of `dist`.

:::note
When running `wrangler deploy`, only your main (entry) Worker will be deployed.

> If using multiple Workers, each must be deployed individually.

Reviewer: ```suggestion
If using multiple Workers, each auxiliary worker must be deployed individually.
```

> You can inspect the `dist` directory and then run `wrangler deploy -c path-to-worker-output-config` for each.

Reviewer: ```suggestion
You can inspect the `dist` directory and then run `wrangler deploy -c dist/<auxiliary-worker>/wrangler.jsonc` for each.
```

:::

> ### `interface AuxiliaryWorkerConfig`

Reviewer: Perhaps here is a good place to do a quick intro to what we mean by auxiliary workers, which we can link to from above?

Author: I felt this was better in the flow of text above.

---
pcx_content_type: reference
title: API

> head: []

Reviewer 2: ```suggestion
```

sidebar:
  order: 3
description: Vite plugin API
---

### `cloudflare()`

The `cloudflare` plugin should be included in the Vite `plugins` array:

```ts {4, 7}
// vite.config.ts

import { defineConfig } from "vite";
import { cloudflare } from "@cloudflare/vite-plugin";

export default defineConfig({
	plugins: [cloudflare()],
});
```

It accepts an optional `PluginConfig` parameter.

### `interface PluginConfig`

> - `configPath?: string`

Reviewer 2: I'd rewrite these to use our `Type` and `MetaInfo` components.
https://developers.cloudflare.com/style-guide/components/type-highlighting/#combined-example

Author: [6bd059e](https://github.com/cloudflare/cloudflare-docs/pull/20481/commits/6bd059e9dd788814191a0c2fb091b11ea15931be)
